forem/forem
# Refactor:Replace Production Check with Average Metrics

## What type of PR is this? (check all applicable)

## Description
Ideally, we would like to minimize the places we are calling `Rails.env.production?` in our codebase because the more we do that, the larger the difference between production and our development environment becomes. The more differences we have, the harder it can be to catch bugs bc we have more code paths that are not being executed when we test locally or in our test suite. This removes a few of those production check statements and replaces them with metrics that gather averages for different counts and scores. 

I changed the SQL statements to be `>=` which then easily handled testing situations where we create only one article, that article is then returned. 

## QA Instructions
Code Path 1
1. Remove your user in development 
2. Set `SiteConfig.suggested_tags = ["beginners", "java", "python"]`
3. Sign up 
4. During onboarding select the 3 tags and click through to the follow suggestion page
5. You should still see user follow suggestions 
Code Path 2
1. Remove your user in development 
2. Sign up 
3. During onboarding DO NOT select any tags and click through to the follow suggestion page
4. You should still see user follow suggestions 

## Added tests?
- [x] Yes/Updated them to be better 

## Added to documentation?


![alt_text](https://media1.tenor.com/images/144c6241cc227e42566060854fd6ea68/tenor.gif?itemid=17779477)
